modules/lang-painless/src/main/java/org/elasticsearch/painless/node/EBinary.java

```diff
@@ -280,6 +285,35 @@ private void analyzeSub(Locals variables) {
         actual = promote;
     }
 
+    private void analyzeFind(Locals variables) {
+        analyzeRegexOp(variables);
+        /*
+         * Constant detection isn't really a thing for FIND because it returns a mutable Matcher. Unless we can detect that it is being
```

Comment:
Would you please explain this comment to me in a bit more depth?  I admit full ignorance here, what is constant detection?